django_irods/views.py

```diff
@@ -380,16 +385,12 @@ def rest_check_task_status(request, task_id, *args, **kwargs):
     '''
     if not task_id:
         task_id = request.POST.get('task_id')
-    result = AsyncResult(task_id)
-    if result.ready():
-        try:
-            ret_value = result.get()
-            return JsonResponse({"status": 'true',
-                                 'payload': str(ret_value)})
-        # use the Broad scope Exception to catch all exception types since this view function can be used for all tasks
-        except Exception:
-            # logging exception will log the full stack trace and prepend a line with the message str input argument
-            logger.exception('An exception is raised from task {}'.format(task_id))
+    task_notification = get_task_notification(task_id)
+    if task_notification:
+        n_status = task_notification['status']
+        if n_status == 'completed':
```

Comment:
@pkdash we could also handle `delivered` and `progress` status.
https://github.com/hydroshare/hydroshare/blob/5045-signal-processor/hs_core/models.py#L4724:L4728
perhaps `delivered` should also -> `true`, same as completed. And `progress` -> `None`

Reply:
nevermind, I went ahead and modified this